### DISCUSSION TITLE
rustdesk change FPS for selfhost？

### DISCUSSION THREAD

**Discussion by himonster1** [rustdesk/rustdesk#3828] [Q&A]
自建了服务器，上传50m带宽肯定够了，但是还是感觉感觉不够丝滑，不够流畅

默认的就算是改成1080p 感觉也就不到30针

**Comment by lamithe** [comment]
我也是，不知道怎么弄，一直稳定在10帧以内。楼主找到办法了吗？

**Reply by 21pages** [reply]
remote menu -> display -> Image Quality -> Custom -> FPS
FPS is also related to encoding/decoding speed,

**Reply by lamithe** [reply]
Thank you, but I setup these as you said. But FPS is still below 10. The GPUs of my remote machine and local machine are both RTX3060 , and I think the performance of GPUs are enough to encode/decode up to 30fps or higher. So how can I fix these?

**Reply by 21pages** [reply]
It's direct connection or relay connection? Trying H264/H265 codec, and what's the resolution?

**Reply by rustdesk** [reply]
When you show a static image, the FPS can be 0

**Reply by lamithe** [reply]
I know. But when I play a video  on remote machine, the local FPS can only be up to 10-12(4K) or 15-20(1080P). My upload bandwidth is 100Mb/s.
And i noticed one thing. In Windows taskmgs.exe, the GPU encode usage is only around 10%, but when I use toDesk, it can be up to 30%(FPS can be 30). So I think if the rustdesk do not use the full resoures of GPU.

**Reply by rustdesk** [reply]
Yes, we are rewriting our hardware codecs implementation. Current hardware codecs is even slower than software codecs, because it does not bypass memory-gpu data copy.

**Reply by lamithe** [reply]
Wow, hope to see the new release soon! Thank you guys!

**Reply by eecoder-dyf** [reply]
If you have problem in improve the performance of hardware codecs, why not just use ffmpeg?

**Comment by aogg** [comment]
+1
